tsc/meetings/2020-03-05.md

```diff
@@ -0,0 +1,104 @@
+
+Agenda:
+
+1) Confirm quorum
+2) Secretary
+3) Release notes build section
+4) Update on Windows changes
+5) Houdini -> VDB voxel size conversion
+6) CI and Github Actions
+7) LevelSet Tracking
+8) Active value deep copy
+9) Volume Advection grain size and topology options
+10) Next meeting
+
+1) Quorum was confirmed.
+
+2) Secretary was Nick Avramoussis
+
+3) Release notes build section
+
+Proposal to separate out the build related release notes from the other
+release note sub-headings. It was noted that these notes have changed
+already in the past to adapt to new formatting being needed. Unanimous
+consensus, with the addition to additionally include any large changes
+in the highlights section. Nick to reformat the current release notes.
+
+4) Update on Windows changes
+
+Nick has been looking through the Windows build and attempting to
+resolve various issues that have been raised. The following issues in
+particular are currently being addressed:
+  #627, #624, #620, #611, #603
+Main issues with the windows build continues to be the ability to
+support the both the Release and Debug builds in a single configuration
+as well as the both static and shared builds. vdb_view continues to be
+unsupported, but the work required to fix it is minimal. Ken to connect
+Nick to community who have reached out expressing desire to help improve
+the windows builds.
+
+5) Houdini -> VDB voxel size conversion
+
+Nick has reported an issue to SideFX regarding conversion of Houdini DOP
+volumes to SOPs and subsequently to VDB volumes. The issue that
+manifests is that a volume which is intrinsically uniform in terms of
+its voxel size components can end up being represented as a non uniform
+scale map when converted to VDB. This can stop various optimizations
+from being performed further down the line and can additionally cause
+unnecessary re-sampling operations to occur. Jeff explained that Houdini
+volumes to not have an explicit representation of their voxel size.
+Instead, The voxel size of a Houdini volume is inferred from its
+transform projected onto a canonical unit cube. This can cause floating
+point differences to manifest when this value is calculated per
+component and presented to the user, as well during conversion to VDB.
+Nick mentioned that an ideal solution would be to be able to query some
+intrinsic details from the Houdini Volume's configuration and use this
+information to determine whether checks should be made on the final
+voxel size computations to ensure they are all equal. Jeff mentioned
+that Houdini Volumes currently contain no such information and instead
+proposed that a clamp can instead be used to ensure components match to
+a given tolerance. Note that whilst this problem originally focused on
+improving the conversion from Houdini->VDB, this issue could also be
+addressed by improving the comparisons between various VDB Map types.
+Nick to verify that the example provided to SideFX exhibits the above
+behaviour.
+
```

Comment:
I found during the meeting that we do clamp the sizes on conversion to VDB, which means it should be a proper uniform-scale map.  However, I think the problem is it might be a *different* uniform scale map frame to frame, so still trigger the resampling problems.  This is where the resampling code should use an almost-equal to not require bitwise identity for inplace operations.

Reply:
So what's the plan of actions here - to fix it in the re-sampling tool by adopting a tolerance?

Reply:
I've created a ticket to track this work here: https://jira.aswf.io/browse/OVDB-140 I've left it unassigned as I won't have time in the near future to pick this back up.